tests/serial/mapping-nearest-projection/helpers.cpp

```diff
@@ -288,21 +313,31 @@ void testQuadMappingNearestProjectionWideKite(bool defineEdgesExplicitly, const
     int idD = interface.setMeshVertex(meshOneID, coordOneD.data());
 
     if (defineEdgesExplicitly) {
+      if (useBulkFunctions) {
+        std::vector ids{idA, idB, idB, idC, idC, idD, idD, idA};
+        interface.setMeshEdges(meshOneID, 4, ids.data());
+      } else {
+        interface.setMeshEdge(meshOneID, idA, idB);
+        interface.setMeshEdge(meshOneID, idB, idC);
+        interface.setMeshEdge(meshOneID, idC, idD);
+        interface.setMeshEdge(meshOneID, idD, idA);
+      }
+    }
 
-      int idAB = interface.setMeshEdge(meshOneID, idA, idB);
-      int idBC = interface.setMeshEdge(meshOneID, idB, idC);
-      int idCD = interface.setMeshEdge(meshOneID, idC, idD);
-      int idDA = interface.setMeshEdge(meshOneID, idD, idA);
-
-      interface.setMeshQuad(meshOneID, idAB, idCD, idBC, idDA);
-
+    if (useBulkFunctions) {
+      std::vector ids{idA, idB, idD, idC};
+      interface.setMeshQuads(meshOneID, 1, ids.data());
     } else {
-      interface.setMeshQuadWithEdges(meshOneID, idA, idB, idD, idC);
+      interface.setMeshQuad(meshOneID, idA, idB, idD, idC);
     }
 
     auto &mesh = testing::WhiteboxAccessor::impl(interface).mesh("MeshOne");
     BOOST_REQUIRE(mesh.vertices().size() == 4);
-    BOOST_REQUIRE(mesh.edges().size() == 5);
+    if (defineEdgesExplicitly) {
+      BOOST_REQUIRE(mesh.edges().size() == 9);
+    } else {
+      BOOST_REQUIRE(mesh.edges().size() == 5);
+    }
```

Comment:
Did we document this behavior somewhere?
This will also have a performance influence for users, no?

Reply:
I extended the documentation of the API functions.
Is it sufficient to mention that the bulk versions are more efficient without stating why?

Reply:
> Is it sufficient to mention that the bulk versions are more efficient without stating why?

Are they actually as you now just call the single versions in a loop? 

But this here is sth different, no?
It's more efficient if you define edges yourself (explicitly) and don't let preCICE do it?

Reply:
Correct, they aren't yet. But they will be.

Upcoming tasks are to prevent duplicates in plural calls and to implement the mesh compression (open PR)

I just wanted to get the API done in this PR.

Reply:
Ah, ok. Then this test will be updated in the mesh compression PR?